### PR TITLE
Convert Windows Services topic to use Powershell

### LINKED ISSUE #10833: Convert Windows Services topic to use Powershell
Add Powershell options for script or command line setup.

There is a New-Service ps command: https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/new-service?view=powershell-6

There is a Set-Acl ps command: https://docs.microsoft.com/en-us/powershell/module/Microsoft.PowerShell.Security/Set-Acl?view=powershell-6

Here is a sample:
$cred = Get-Credential

$currentDirectory = (Get-Item -Path ".\").FullName

$binPath = $currentDirectory + '\YourService.exe --environment Production --server.urls "http://*:80"'

$acl = Get-Acl $currentDirectory

$aclRuleArgs = $cred.UserName, "Read,Write,ReadAndExecute", "ContainerInherit, ObjectInherit", "None", "Allow"

$AccessRule = New-Object System.Security.AccessControl.FileSystemAccessRule $aclRuleArgs

$acl.SetAccessRule($AccessRule)

$acl | Set-Acl $currentDirectory

New-Service -Name {SERVICE NAME} -BinaryPathName "$binPath" -Credential $cred -DisplayName {SERVICE DISPLAYNAME} -StartupType Automatic -Description {SERVICE DESCRIPTION}

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b0bd8cb8-7e28-8c4d-a550-3aea8f3ab4dd
* Version Independent ID: 4b60d24e-ef05-15e5-10b4-483094081bf3
* Content: [Host ASP.NET Core in a Windows Service](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/windows-service?view=aspnetcore-2.2#feedback)
* Content Source: [aspnetcore/host-and-deploy/windows-service.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/host-and-deploy/windows-service.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **tdykstra**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #10833 

Thanks @pherbel ... I used your ACL bits.

Notes:

* We need PS6 for this: Not only is it the latest, the `Remove-Service` cmdlet didn't become available until PS6, and that's a *nice-to-have* here.
* WRT granting *Log on as a service* rights to the user: We can't use the UserRights script:
  * That script doesn't work in PS6.
  * It's not supported by MS anyway.
  
  However, we certainly can use a security template as I do here. @blowdart will be *The First Decider* on what we ultimately do.
* There's no OOB way AFAICT to create a local user in PS6. For example, the `New-LocalUser` command of PS5.1 doesn't work. I stick with the `net user` command here.

### CODE REVIEW

**Comment by blowdart** [comment]
As I don't know powershell that well @TravisEz13 might have some thoughts about it.

**Comment by guardrex** [comment]
@blowdart What about the use of the security template and `secedit.exe` command to give the user *Log on as a service* privileges? It's supported and it's easy to show-'n-tell for the reader to perform. There's no supported PS way to do it ... not as far as I can tell anyway.

**Comment by blowdart** [comment]
That's certainly novel :)

**Comment by guardrex** [comment]
Well ... it's not as bad as my usual 😝 *RexHacks* 😝 around here. @shirhatti may have one or more alternatives. This is fairly easy to grok/perform tho. Seems to work well.

**Comment by TravisEz13** [comment]
`New-LocalUser` should work in PS6.2 on windows.

**Comment by guardrex** [comment]
I see ... I was using 6.1.3. They seem to be saying that that's the latest. Is that correct? Is 6.2 in preview?

**Comment by guardrex** [comment]
Yeah ... I see it. It's in RC. I'll open an issue to replace that bit when 6.2 RTMs.

**Comment by TravisEz13** [comment]
I just verified that New-LocalUser works at least when you don't let it prompt for a password.  I'll have to open an issue on the prompt, but that may never get fixed.

**Comment by TravisEz13** [comment]
Can you point me to the `UserRights` script?

**Comment by guardrex** [comment]
Cool ... I'll take care of swapping for the `net user` command when 6.2 RTMs.

Here it is ... https://gallery.technet.microsoft.com/scriptcenter/Grant-Revoke-Query-user-26e259b0

It broke when I tried to use it. However, it isn't supported (unless management wants to give me a pass here ... to cover my b-hind :smile:).

**Comment by TravisEz13** [comment]
I think that script can be fixed (not making any claims about supportability or maintainability.  Actually, I believe it has similar issue to why `New-LocalUser` had to be pulled from PowerShell 6)

It's actually all C# code that is broken (So far), which is being compiled at runtime in PowerShell.  This is not a great design for cross framework compatibility.

I'm not sure I have time, but if someone want to work on it, my recommendation would be to convert it to a Net/PowerShell Standard/Windows Compat based C# PowerShell module.  This would work on Windows PowerShell (in Windows 10) and PowerShell Core.

**Comment by guardrex** [comment]
Thanks @TravisEz13

I'll have an :ear: open for @shirhatti's feedback.

**Comment by guardrex** [comment]
Ping @shirhatti - This one is also fairly short.

The main concern with the updates here is granting the *Log on as a service* right to the service account.

This PR includes a move from *icacls.exe* to *secedit.exe* (with a security template), which is fairly pointless. Now that I've :zzz:🛌:zzz: on it, I like the *icacls.exe* approach better.

[UserRights](https://gallery.technet.microsoft.com/scriptcenter/Grant-Revoke-Query-user-26e259b0) doesn't work with PS6 and imo presents significant security risks.

Does engineering want to come up with a PS script for this step (akin to the UserRights approach) that we can trust and include with the topic? Want me to hack something together as a first pass? ... good for a laugh anyway :smile:.

**Comment by shirhatti** [comment]
Maybe I missed the memo on this, but why are we converting this doc to use Powershell?

**Comment by guardrex** [comment]
https://github.com/aspnet/Docs/pull/10837#pullrequestreview-201724257

We can close this if you want. In addition to the inability to grant *Log on as a service*, `New-LocalUser` doesn't return until 6.2 RTM (it's RC right now; I opened https://github.com/aspnet/Docs/issues/11344 to cover it). If we do close the PR, do you want to close the issue, as well, or do you just want me to push the issue to the Backlog? (If we close both the PR **_and the issue_**, I won't bill MS for any of this work.)

**Comment by shirhatti** [comment]
Since @blowdart asked for it, I'll let him make the call.

File: aspnetcore/host-and-deploy/windows-service.md [hunk 1]
```diff
@@ -165,106 +169,76 @@ For more information, see [Service User Accounts](/windows/desktop/services/serv
...
+   secedit /configure /db secedit.sdb /cfg "c:\Users\guard\Desktop\LOCAL_SECURITY_TEMPLATE.inf"
+   ```
 
-### Create the service
+## Create the service
 
-Use the [sc.exe](https://technet.microsoft.com/library/bb490995) command-line tool to create the service from an administrative command shell. The `binPath` value is the path to the app's executable, which includes the executable file name. **The space between the equal sign and the quote character of each parameter and value is required.**
+Use the [RegisterService.ps1](https://github.com/aspnet/Docs/tree/master/aspnetcore/host-and-deploy/windows-service/scripts) PowerShell script to register the service. From an administrative PowerShell 6 command prompt, execute the following command:
 
 ```console
```
Comment (shirhatti) [inline aspnetcore/host-and-deploy/windows-service.md]: nit: s/console/powershell

File: aspnetcore/host-and-deploy/windows-service.md [hunk 2]
```diff
@@ -165,106 +169,76 @@ For more information, see [Service User Accounts](/windows/desktop/services/serv
...
 In the following example for the sample app:
 
 * The service is named **MyService**.
-* The published service resides in the *c:\\svc* folder. The app executable is named *SampleApp.exe*. Enclose the `binPath` value in double quotation marks (").
-* The service runs under the `ServiceUser` account. Replace `{DOMAIN}` with the user account's domain or local machine name. Enclose the `obj` value in double quotation marks ("). Example: If the hosting system is a local machine named `MairaPC`, set `obj` to `"MairaPC\ServiceUser"`.
-* Replace `{PASSWORD}` with the user account's password. Enclose the `password` value in double quotation marks (").
+* The published service resides in the *c:\\svc* folder. The app executable is named *SampleApp.exe*.
+* The service runs under the `ServiceUser` account. In the following example, the local machine name is `Desktop-PC`.
 
 ```console
```
Comment (shirhatti) [inline aspnetcore/host-and-deploy/windows-service.md]: nit: s/console/powershell

**Comment by guardrex** [comment]
@blowdart Ur call ... which way do u want to go? ...

* *icacls.exe*
* *secedit.exe* (with a security template)

I'm leaning back toward what we have in the topic today (*icacls.exe*) just because it doesn't require the extra *template* bit. It's just a command that they can execute in the shell.

**Comment by guardrex** [comment]
... and just note to all that we're not going to make a PS script today for the *Log on as a service* permission, but I would like to reconsider it in the future. I'm going to open a backlog issue (P3) so that we can take another look down the road at it.

**Comment by blowdart** [comment]
icacls please

**Pull request merged**